hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/ChunkUtils.java

```diff
@@ -227,10 +248,40 @@ public static void readData(File file, ByteBuffer[] buffers,
         bytesRead, offset, file);
 
     validateReadSize(len, bytesRead);
+  }
 
-    for (ByteBuffer buf : buffers) {
-      buf.flip();
-    }
+  public static ChunkBufferToByteString readData(File file, long chunkSize,
+      long offset, long length, HddsVolume volume)
+      throws StorageContainerException {
+    final List<ByteBuf> buffers = ChunkUtils.readDataNettyChunkedNioFile(
+        file, Math.toIntExact(chunkSize), offset, length, volume);
+    return ChunkBufferToByteString.wrap(buffers);
+  }
+
+  /**
+   * Read data from the given file using {@link ChunkedNioFile}.
+   *
+   * @return a list of {@link ByteBuf} containing the data.
+   */
+  private static List<ByteBuf> readDataNettyChunkedNioFile(
+      File file, int chunkSize,
+      long offset, long length, HddsVolume volume)
+      throws StorageContainerException {
+
+    final List<ByteBuf> buffers = new ArrayList<>(
+        Math.toIntExact((length - 1) / chunkSize) + 1);
+    readData(file, offset, length, volume, channel -> {
+      final ChunkedNioFile f = new ChunkedNioFile(
+          channel, offset, length, chunkSize);
+      long readLen = 0;
+      while (readLen < length) {
+        final ByteBuf buf = f.readChunk(PooledByteBufAllocator.DEFAULT);
```

Comment:
We should file a jira to allow for the allocator to be configured to play with the settings available.

Reply:
Actually since we are trying to read chunks here it might make sense to play with the allocator to select buffers that are the same as chunk size.

Reply:
Currently, the allocator will use `chunkSize` as the buffer size.